diff --git a/node_modules/@2060.io/credo-ts-didcomm-receipts/build/index.mjs b/node_modules/@2060.io/credo-ts-didcomm-receipts/build/index.mjs
index f1132cf..d8bee13 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-receipts/build/index.mjs
+++ b/node_modules/@2060.io/credo-ts-didcomm-receipts/build/index.mjs
@@ -158,7 +158,7 @@ let DidCommReceiptsService = class DidCommReceiptsService$1 {
 		});
 	}
 };
-DidCommReceiptsService = __decorate([scoped(Lifecycle.ContainerScoped), __decorateMetadata("design:paramtypes", [typeof (_ref$1 = typeof EventEmitter !== "undefined" && EventEmitter) === "function" ? _ref$1 : Object])], DidCommReceiptsService);
+DidCommReceiptsService = __decorate([injectable(), scoped(Lifecycle.ContainerScoped), __decorateMetadata("design:paramtypes", [typeof (_ref$1 = typeof EventEmitter !== "undefined" && EventEmitter) === "function" ? _ref$1 : Object])], DidCommReceiptsService);
 
 //#endregion
 //#region src/DidCommReceiptsApi.ts
