diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/index.mjs b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/index.mjs
index 187bacf..bd438f5 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/index.mjs
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/index.mjs
@@ -160,6 +160,7 @@ let UserProfileRepository = class UserProfileRepository$1 extends Repository {
 	}
 };
 UserProfileRepository = __decorate([
+	injectable(),
 	scoped(Lifecycle.ContainerScoped),
 	__decorateParam(0, inject(InjectionSymbols.StorageService)),
 	__decorateMetadata("design:paramtypes", [Object, typeof (_ref$2 = typeof EventEmitter !== "undefined" && EventEmitter) === "function" ? _ref$2 : Object])
@@ -284,7 +285,7 @@ let DidCommUserProfileService = class DidCommUserProfileService$1 {
 		});
 	}
 };
-DidCommUserProfileService = __decorate([scoped(Lifecycle.ContainerScoped), __decorateMetadata("design:paramtypes", [
+DidCommUserProfileService = __decorate([injectable(), scoped(Lifecycle.ContainerScoped), __decorateMetadata("design:paramtypes", [
 	typeof (_ref$1 = typeof UserProfileRepository !== "undefined" && UserProfileRepository) === "function" ? _ref$1 : Object,
 	typeof (_ref2$1 = typeof DidCommConnectionService !== "undefined" && DidCommConnectionService) === "function" ? _ref2$1 : Object,
 	typeof (_ref3$1 = typeof EventEmitter !== "undefined" && EventEmitter) === "function" ? _ref3$1 : Object
