diff --git a/node_modules/@2060.io/credo-ts-didcomm-shorten-url/build/index.mjs b/node_modules/@2060.io/credo-ts-didcomm-shorten-url/build/index.mjs
index fd8d565..8e80644 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-shorten-url/build/index.mjs
+++ b/node_modules/@2060.io/credo-ts-didcomm-shorten-url/build/index.mjs
@@ -254,6 +254,7 @@ let DidCommShortenUrlRepository = class DidCommShortenUrlRepository$1 extends Re
 	}
 };
 DidCommShortenUrlRepository = __decorate([
+	injectable(),
 	scoped(Lifecycle.ContainerScoped),
 	__decorateParam(0, inject(InjectionSymbols.StorageService)),
 	__decorateMetadata("design:paramtypes", [Object, typeof (_ref$2 = typeof EventEmitter !== "undefined" && EventEmitter) === "function" ? _ref$2 : Object])
