diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.d.ts b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.d.ts
index 2647649..9103c0b 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.d.ts
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.d.ts
@@ -37,7 +37,7 @@ export declare class UserProfileApi {
      *
      * @returns updated User Profile data
      */
-    updateUserProfileData(props: Partial<UserProfileData>): Promise<UserProfileData>;
+    updateUserProfileData(props: Partial<Omit<UserProfileData,'updatedAt'>>): Promise<UserProfileData>;
     /**
      * Retrieve our User Profile Data from storage.
      *
diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
index bad2e56..e59d672 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
@@ -84,6 +84,7 @@ let UserProfileApi = class UserProfileApi {
             displayIcon: userProfile.displayIcon,
             description: userProfile.description,
             preferredLanguage: userProfile.preferredLanguage,
+            updatedAt: userProfile.updatedAt
         };
     }
 };
diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
index 5777d06..2ef1046 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
@@ -9,4 +9,5 @@ export interface UserProfileData {
     displayIcon?: PictureData;
     description?: string;
     preferredLanguage?: string;
+    updatedAt?: Date;
 }
