diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
index bad2e56..e59d672 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/UserProfileApi.js
@@ -84,6 +84,7 @@ let UserProfileApi = class UserProfileApi {
             displayIcon: userProfile.displayIcon,
             description: userProfile.description,
             preferredLanguage: userProfile.preferredLanguage,
+            updatedAt: userProfile.updatedAt
         };
     }
 };
diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
index 5777d06..2ef1046 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/model/UserProfile.d.ts
@@ -9,4 +9,5 @@ export interface UserProfileData {
     displayIcon?: PictureData;
     description?: string;
     preferredLanguage?: string;
+    updatedAt?: Date;
 }
diff --git a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/services/UserProfileService.js b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/services/UserProfileService.js
index 2f6ce7d..b5e9192 100644
--- a/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/services/UserProfileService.js
+++ b/node_modules/@2060.io/credo-ts-didcomm-user-profile/build/services/UserProfileService.js
@@ -36,6 +36,7 @@ let UserProfileService = class UserProfileService {
             displayName: userProfile.displayName,
             displayPicture: userProfile.displayPicture,
             preferredLanguage: userProfile.preferredLanguage,
+            updatedAt: userProfile.updatedAt
         };
         Object.assign(userProfile, props);
         await this.userProfileRepository.update(agentContext, userProfile);
