diff --git a/android/src/main/java/com/reactnativeimageresizer/ImageResizer.java b/android/src/main/java/com/reactnativeimageresizer/ImageResizer.java
index d148265..d39bbf8 100644
--- a/android/src/main/java/com/reactnativeimageresizer/ImageResizer.java
+++ b/android/src/main/java/com/reactnativeimageresizer/ImageResizer.java
@@ -21,6 +21,7 @@ import java.io.IOException;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.util.Date;
+import java.util.Objects;
 
 /**
  * Provide methods to resize and rotate an image file.
@@ -319,8 +320,7 @@ public class ImageResizer {
     try {
       // ExifInterface(InputStream) only exists since Android N (r24)
       if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-        InputStream input = context.getContentResolver().openInputStream(uri);
-        ExifInterface ei = new ExifInterface(input);
+        ExifInterface ei = new ExifInterface(Objects.requireNonNull(uri.getPath()));
         return getOrientationMatrix(ei);
       }
       File file = getFileFromUri(context, uri);
