diff --git a/node_modules/@credo-ts/askar/build/AskarApi.mjs b/node_modules/@credo-ts/askar/build/AskarApi.mjs
index 954ba89..b004069 100644
--- a/node_modules/@credo-ts/askar/build/AskarApi.mjs
+++ b/node_modules/@credo-ts/askar/build/AskarApi.mjs
@@ -2,8 +2,7 @@ import { AskarModuleConfig } from "./AskarModuleConfig.mjs";
 import { __decorateMetadata } from "./_virtual/_@oxc-project_runtime@0.99.0/helpers/decorateMetadata.mjs";
 import { __decorate } from "./_virtual/_@oxc-project_runtime@0.99.0/helpers/decorate.mjs";
 import { AskarStoreManager } from "./AskarStoreManager.mjs";
-import { AgentContext } from "@credo-ts/core";
-import { injectable as injectable$1 } from "tsyringe";
+import { AgentContext, injectable } from "@credo-ts/core";
 
 //#region src/AskarApi.ts
 var _ref, _ref2, _ref3;
@@ -83,7 +82,7 @@ let AskarApi = class AskarApi$1 {
 		await this.askarStoreManager.closeStore(this.agentContext);
 	}
 };
-AskarApi = __decorate([injectable$1(), __decorateMetadata("design:paramtypes", [
+AskarApi = __decorate([injectable(), __decorateMetadata("design:paramtypes", [
 	typeof (_ref = typeof AgentContext !== "undefined" && AgentContext) === "function" ? _ref : Object,
 	typeof (_ref2 = typeof AskarStoreManager !== "undefined" && AskarStoreManager) === "function" ? _ref2 : Object,
 	typeof (_ref3 = typeof AskarModuleConfig !== "undefined" && AskarModuleConfig) === "function" ? _ref3 : Object
diff --git a/node_modules/@credo-ts/askar/build/AskarStoreManager.mjs b/node_modules/@credo-ts/askar/build/AskarStoreManager.mjs
index 89c0637..257a3f6 100644
--- a/node_modules/@credo-ts/askar/build/AskarStoreManager.mjs
+++ b/node_modules/@credo-ts/askar/build/AskarStoreManager.mjs
@@ -15,9 +15,8 @@ import "./utils/index.mjs";
 import { __decorateMetadata } from "./_virtual/_@oxc-project_runtime@0.99.0/helpers/decorateMetadata.mjs";
 import { __decorateParam } from "./_virtual/_@oxc-project_runtime@0.99.0/helpers/decorateParam.mjs";
 import { __decorate } from "./_virtual/_@oxc-project_runtime@0.99.0/helpers/decorate.mjs";
-import { AgentContext, InjectionSymbols, JsonTransformer, StorageVersionRecord } from "@credo-ts/core";
+import { AgentContext, InjectionSymbols, JsonTransformer, StorageVersionRecord, inject, injectable } from "@credo-ts/core";
 import { KdfMethod, Session, Store, StoreKeyMethod } from "@openwallet-foundation/askar-shared";
-import { inject, injectable as injectable$1 } from "tsyringe";
 
 //#region src/AskarStoreManager.ts
 var _ref;
@@ -376,7 +375,7 @@ let AskarStoreManager = class AskarStoreManager$1 {
 	}
 };
 AskarStoreManager = __decorate([
-	injectable$1(),
+	injectable(),
 	__decorateParam(0, inject(InjectionSymbols.FileSystem)),
 	__decorateMetadata("design:paramtypes", [Object, typeof (_ref = typeof AskarModuleConfig !== "undefined" && AskarModuleConfig) === "function" ? _ref : Object])
 ], AskarStoreManager);
