diff --git a/node_modules/@credo-ts/react-native/build/rnfs.mjs b/node_modules/@credo-ts/react-native/build/rnfs.mjs
index 1a2eb7a..94c6c08 100644
--- a/node_modules/@credo-ts/react-native/build/rnfs.mjs
+++ b/node_modules/@credo-ts/react-native/build/rnfs.mjs
@@ -1,12 +1,12 @@
-import { __require } from "./_virtual/_rolldown/runtime.mjs";
+
 
 //#region src/rnfs.ts
 const rnfs = () => {
 	try {
-		return __require("@dr.pogodin/react-native-fs");
+		return require("@dr.pogodin/react-native-fs");
 	} catch {
 		try {
-			return __require("react-native-fs");
+			return require("react-native-fs");
 		} catch {
 			throw new Error(`Could not find '@dr.pogodin/react-native-fs' or 'react-native-fs' as a peerDependency. Make sure to add either one of them`);
 		}
