diff --git a/node_modules/@credo-ts/didcomm/build/DidCommDispatcher.mjs b/node_modules/@credo-ts/didcomm/build/DidCommDispatcher.mjs
index e1a0c77..804363d 100644
--- a/node_modules/@credo-ts/didcomm/build/DidCommDispatcher.mjs
+++ b/node_modules/@credo-ts/didcomm/build/DidCommDispatcher.mjs
@@ -29,6 +29,16 @@ let DidCommDispatcher = class DidCommDispatcher {
 			if (!messageHandler) throw new DidCommProblemReportError(`Error handling message ${inboundMessageContext.message.id} with type ${inboundMessageContext.message.type}. The message type is not supported`, { problemCode: DidCommProblemReportReason.MessageParseFailure });
 			const outboundMessage = await messageHandler.handle(inboundMessageContext);
 			if (outboundMessage) inboundMessageContext.setResponseMessage(outboundMessage);
+
+			this.eventEmitter.emit(inboundMessageContext.agentContext, {
+			type: DidCommEventTypes.DidCommMessageProcessed,
+			payload: {
+				message: inboundMessageContext.message,
+				connection: inboundMessageContext.connection,
+				receivedAt: inboundMessageContext.receivedAt,
+				encryptedMessage: inboundMessageContext.encryptedMessage
+			}
+			});
 			await next();
 		};
 		this.messageSender = messageSender;
@@ -73,15 +83,6 @@ let DidCommDispatcher = class DidCommDispatcher {
 			if (!outboundMessage.inboundMessageContext) outboundMessage.inboundMessageContext = messageContext;
 			await this.messageSender.sendMessage(outboundMessage);
 		}
-		this.eventEmitter.emit(agentContext, {
-			type: DidCommEventTypes.DidCommMessageProcessed,
-			payload: {
-				message,
-				connection,
-				receivedAt: messageContext.receivedAt,
-				encryptedMessage
-			}
-		});
 	}
 };
 DidCommDispatcher = __decorate([
