diff --git a/node_modules/@hyperledger/aries-askar-react-native/android/CMakeLists.txt b/node_modules/@hyperledger/aries-askar-react-native/android/CMakeLists.txt
index 74b6aff..5b2a966 100644
--- a/node_modules/@hyperledger/aries-askar-react-native/android/CMakeLists.txt
+++ b/node_modules/@hyperledger/aries-askar-react-native/android/CMakeLists.txt
@@ -87,7 +87,14 @@ if(${REACT_NATIVE_VERSION} LESS 71)
   )
 endif()
 
-if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 76)
+  target_link_libraries(
+    ${PACKAGE_NAME}
+    ReactAndroid::jsi
+    ReactAndroid::reactnative
+    fbjni::fbjni
+  )
+elseif(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
   target_link_libraries(
     ${PACKAGE_NAME}
     ReactAndroid::jsi
diff --git a/node_modules/@hyperledger/aries-askar-react-native/aries-askar.podspec b/node_modules/@hyperledger/aries-askar-react-native/aries-askar.podspec
index 655864c..babed58 100644
--- a/node_modules/@hyperledger/aries-askar-react-native/aries-askar.podspec
+++ b/node_modules/@hyperledger/aries-askar-react-native/aries-askar.podspec
@@ -16,7 +16,8 @@ Pod::Spec.new do |s|
   s.header_mappings_dir = "cpp"
 
   s.pod_target_xcconfig = {
-    :USE_HEADERMAP => "No"
+    :USE_HEADERMAP => "No",
+    'CLANG_CXX_LANGUAGE_STANDARD' => "c++20",
   }
 
   s.ios.vendored_frameworks = "native/mobile/ios/aries_askar.xcframework"
