diff --git a/node_modules/@notifee/react-native/dist/types/Module.d.ts b/node_modules/@notifee/react-native/dist/types/Module.d.ts
index 05edb54..ed19de2 100644
--- a/node_modules/@notifee/react-native/dist/types/Module.d.ts
+++ b/node_modules/@notifee/react-native/dist/types/Module.d.ts
@@ -13,8 +13,12 @@ export interface Module {
      *
      * This method does not cancel Android [Foreground Service](/react-native/docs/android/foreground-service)
      * notifications.
+     * @param notificationIds An array of notifications IDs. This is automatically generated and returned
+     * when creating a notification, or has been set manually via the `id` property.
+     *
+     * @param tag The tag set when creating the notification. This is only relative to Android.
      */
-    cancelAllNotifications(notificationIds?: string[]): Promise<void>;
+    cancelAllNotifications(notificationIds?: string[], tag?: string): Promise<void>;
     /**
      * API used to cancel any displayed notifications.
      *
