diff --git a/node_modules/@openwallet-foundation/askar-react-native/cpp/askar.cpp b/node_modules/@openwallet-foundation/askar-react-native/cpp/askar.cpp
index f823c33..583c937 100644
--- a/node_modules/@openwallet-foundation/askar-react-native/cpp/askar.cpp
+++ b/node_modules/@openwallet-foundation/askar-react-native/cpp/askar.cpp
@@ -312,7 +312,7 @@ jsi::Value storeCopyProfile(jsi::Runtime &rt, jsi::Object options) {
   state->rt = &rt;
 
   ErrorCode code = askar_store_copy_profile(
-      fromHandle, toHandle, fromProfile.c_str(), toProfile.c_str(), callbackWithResponse, CallbackId(state));
+      fromHandle, toHandle, fromProfile.c_str(), toProfile.c_str(), callback, CallbackId(state));
 
   return createReturnValue(rt, code, nullptr);
 }
