diff --git a/node_modules/@verana-labs/verre/build/cjs/resolver/didValidator.js b/node_modules/@verana-labs/verre/build/cjs/resolver/didValidator.js
index 449bd18..6a819ac 100644
--- a/node_modules/@verana-labs/verre/build/cjs/resolver/didValidator.js
+++ b/node_modules/@verana-labs/verre/build/cjs/resolver/didValidator.js
@@ -228,6 +228,7 @@ async function processDidDocument(did, didDocument, agentContext, verifiablePubl
     await Promise.all(didDocument.service.map(async (didService) => {
         switch (didService.type) {
             case 'LinkedVerifiablePresentation': {
+                if (!didService.id.includes('c-vp')) return
                 const vp = await resolveServiceVP(didService);
                 if (!vp)
                     throw new utils_1.TrustError(types_1.TrustErrorCode.NOT_SUPPORTED, `Invalid Linked Verifiable Presentation for service id: '${didService.id}'`);
diff --git a/node_modules/@verana-labs/verre/build/resolver/didValidator.js b/node_modules/@verana-labs/verre/build/resolver/didValidator.js
index 096dea0..11d9a96 100644
--- a/node_modules/@verana-labs/verre/build/resolver/didValidator.js
+++ b/node_modules/@verana-labs/verre/build/resolver/didValidator.js
@@ -189,6 +189,7 @@ async function processDidDocument(did, didDocument, agentContext, verifiablePubl
     await Promise.all(didDocument.service.map(async (didService) => {
         switch (didService.type) {
             case 'LinkedVerifiablePresentation': {
+                if (!didService.id.includes('c-vp')) return
                 const vp = await resolveServiceVP(didService);
                 if (!vp)
                     throw new TrustError(TrustErrorCode.NOT_SUPPORTED, `Invalid Linked Verifiable Presentation for service id: '${didService.id}'`);
