diff --git a/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/DataGroup14.swift b/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/DataGroup14.swift
index 6be29e9..f96a426 100644
--- a/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/DataGroup14.swift
+++ b/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/DataGroup14.swift
@@ -23,7 +23,7 @@ public class DataGroup14 : DataGroup {
     }
     
     override func parse(_ data: [UInt8]) throws {
-        let p = SimpleASN1DumpParser()
+        /*let p = SimpleASN1DumpParser()
         asn1 = try p.parse(data: Data(body))
         
         // Bit of a hack at the moment - passing in the body - if we had a decent ASN1 parser then this would be better! ;)
@@ -32,6 +32,6 @@ public class DataGroup14 : DataGroup {
                let secInfo = SecurityInfo.getInstance( object:child, body : body ) {
                 securityInfos.append(secInfo)
             }
-        }
+        }*/
     }
 }
diff --git a/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/SOD.swift b/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/SOD.swift
index 6bbb254..4578af2 100644
--- a/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/SOD.swift
+++ b/node_modules/react-native-eid-reader/ios/NFCPassportReader/DataGroups/SOD.swift
@@ -80,8 +80,8 @@ class SOD : DataGroup {
     }
 
     override func parse(_ data: [UInt8]) throws {
-        let p = SimpleASN1DumpParser()
-        asn1 = try p.parse(data: Data(body))
+        //let p = SimpleASN1DumpParser()
+        //asn1 = try p.parse(data: Data(body))
     }
     
     /// Returns the public key from the embedded X509 certificate
diff --git a/node_modules/react-native-eid-reader/ios/NFCPassportReader/PassportReader.swift b/node_modules/react-native-eid-reader/ios/NFCPassportReader/PassportReader.swift
index 773f469..d0eb956 100644
--- a/node_modules/react-native-eid-reader/ios/NFCPassportReader/PassportReader.swift
+++ b/node_modules/react-native-eid-reader/ios/NFCPassportReader/PassportReader.swift
@@ -253,7 +253,9 @@ extension PassportReader {
         self.readerSession?.invalidate()
 
         // If we have a masterlist url set then use that and verify the passport now
-        self.passport.verifyPassport(masterListURL: self.masterListURL, useCMSVerification: self.passiveAuthenticationUsesOpenSSL)
+        if (self.masterListURL != nil) { 
+            self.passport.verifyPassport(masterListURL: self.masterListURL, useCMSVerification: self.passiveAuthenticationUsesOpenSSL)
+        }
 
         return self.passport
     }
