diff --git a/node_modules/react-native-share-menu/android/build.gradle b/node_modules/react-native-share-menu/android/build.gradle
index 9557fdb..cc17e65 100644
--- a/node_modules/react-native-share-menu/android/build.gradle
+++ b/node_modules/react-native-share-menu/android/build.gradle
@@ -1,12 +1,20 @@
 apply plugin: 'com.android.library'
 
+def getExtOrDefault(name) {
+    return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['ReactNativeShareMenu_' + name]
+}
+
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['ReactNativeShareMenu_' + name]).toInteger()
+}
+
 android {
-    compileSdkVersion 29
-    buildToolsVersion "29.0.2"
+    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+    buildToolsVersion getExtOrDefault('buildToolsVersion')
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 29
+        minSdkVersion getExtOrIntegerDefault('minSdkVersion')
+        targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
         versionCode 1
         versionName "1.0"
         ndk {
@@ -22,3 +30,4 @@ dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
 }
+
diff --git a/node_modules/react-native-share-menu/android/gradle.properties b/node_modules/react-native-share-menu/android/gradle.properties
new file mode 100644
index 0000000..8b3e26e
--- /dev/null
+++ b/node_modules/react-native-share-menu/android/gradle.properties
@@ -0,0 +1,4 @@
+ReactNativeShareMenu_compileSdkVersion=34
+ReactNativeShareMenu_buildToolsVersion=34.0.0
+ReactNativeShareMenu_minSdkVersion=25
+ReactNativeShareMenu_targetSdkVersion=34
\ No newline at end of file
diff --git a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
index 09abd7b..947c19f 100644
--- a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+++ b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
@@ -13,8 +13,10 @@ import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 
 import android.app.Activity;
+import android.content.ContentResolver;
 import android.content.Intent;
 import android.net.Uri;
+import android.webkit.MimeTypeMap;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
@@ -56,30 +58,42 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
     String action = intent.getAction();
 
     WritableMap data = Arguments.createMap();
-    data.putString(MIME_TYPE_KEY, type);
+    WritableArray uriArr = Arguments.createArray();
 
     if (Intent.ACTION_SEND.equals(action)) {
-      if ("text/plain".equals(type)) {
-        data.putString(DATA_KEY, intent.getStringExtra(Intent.EXTRA_TEXT));
-        return data;
-      }
-
-      Uri fileUri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
-      if (fileUri != null) {
-        data.putString(DATA_KEY, fileUri.toString());
-        return data;
-      }
-    } else if (Intent.ACTION_SEND_MULTIPLE.equals(action)) {
-      ArrayList<Uri> fileUris = intent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
-      if (fileUris != null) {
-        WritableArray uriArr = Arguments.createArray();
-        for (Uri uri : fileUris) {
-          uriArr.pushString(uri.toString());
+        WritableMap dataRow = Arguments.createMap();
+        if ("text/plain".equals(type)) {
+          dataRow.putString(MIME_TYPE_KEY, type);
+          dataRow.putString(DATA_KEY, intent.getStringExtra(Intent.EXTRA_TEXT));
+          uriArr.pushMap(dataRow);
+          data.putArray(DATA_KEY, uriArr);
+          return data;
+        }
+        Uri fileUri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+        if (fileUri != null) {
+          ContentResolver cr = getReactApplicationContext().getContentResolver();
+          String mimeType = cr.getType(fileUri);
+          dataRow.putString(MIME_TYPE_KEY, mimeType);
+          dataRow.putString(DATA_KEY,  fileUri.toString());
+          uriArr.pushMap(dataRow);
+          data.putArray(DATA_KEY, uriArr);
+          return data;
+        }
+      } else if (Intent.ACTION_SEND_MULTIPLE.equals(action)) {
+        ArrayList<Uri> fileUris = intent.getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+        if (fileUris != null) {
+          for (Uri uri : fileUris) {
+            WritableMap dataRow = Arguments.createMap();
+            ContentResolver cr = getReactApplicationContext().getContentResolver();
+            String mimeType = cr.getType(uri);
+            dataRow.putString(MIME_TYPE_KEY, mimeType);
+            dataRow.putString(DATA_KEY, uri.toString());
+            uriArr.pushMap(dataRow);
+            }
+            data.putArray(DATA_KEY, uriArr);
+            return data;
         }
-        data.putArray(DATA_KEY, uriArr);
-        return data;
       }
-    }
 
     return null;
   }
diff --git a/node_modules/react-native-share-menu/index.d.ts b/node_modules/react-native-share-menu/index.d.ts
new file mode 100644
index 0000000..da01325
--- /dev/null
+++ b/node_modules/react-native-share-menu/index.d.ts
@@ -0,0 +1,30 @@
+type SharedItem = {
+  mimeType: string
+  data: string
+}
+
+export type SharedData = Record<'data', SharedItem[]>
+
+export type ShareCallback = (share?: SharedData) => void;
+
+export interface ShareListener {
+    remove(): void;
+}
+
+interface ShareMenu {
+    getSharedText(callback: ShareCallback): void;
+    getInitialShare(callback: ShareCallback): void;
+    addNewShareListener(callback: ShareCallback): ShareListener;
+    clearSharedText(): void;
+}
+
+interface ShareMenuReactView {
+    dismissExtension(error?: string): void;
+    openApp(): void;
+    continueInApp(extraData?: object): void;
+    data(): Promise<{ mimeType: string; data: string }>;
+}
+
+export const ShareMenuReactView: ShareMenuReactView;
+declare const ShareMenu: ShareMenu;
+export default ShareMenu;
diff --git a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
index e290cce..7380ee4 100644
--- a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+++ b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
@@ -147,7 +147,7 @@ public class ShareMenuReactView: NSObject {
                                         // Writing the image to the URL
                                         try imageData.write(to: imageURL)
 
-                                        results.append([DATA_KEY: imageUrl.absoluteString, MIME_TYPE_KEY: imageURL.extractMimeType()])
+                                        results.append([DATA_KEY: imageURL.absoluteString, MIME_TYPE_KEY: imageURL.extractMimeType()])
                                     } catch {
                                         callback(nil, NSException(name: NSExceptionName(rawValue: "Error"), reason:"Can't load image", userInfo:nil))
                                     }
diff --git a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
index f42bce6..ee36062 100644
--- a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
+++ b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
@@ -13,7 +13,7 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
   func sourceURL(for bridge: RCTBridge!) -> URL! {
 #if DEBUG
     return RCTBundleURLProvider.sharedSettings()?
-      .jsBundleURL(forBundleRoot: "index.share", fallbackResource: nil)
+      .jsBundleURL(forBundleRoot: "index.share")
 #else
     return Bundle.main.url(forResource: "main", withExtension: "jsbundle")
 #endif
diff --git a/node_modules/react-native-share-menu/ios/ShareViewController.swift b/node_modules/react-native-share-menu/ios/ShareViewController.swift
index 12d8c92..74367cf 100644
--- a/node_modules/react-native-share-menu/ios/ShareViewController.swift
+++ b/node_modules/react-native-share-menu/ios/ShareViewController.swift
@@ -33,6 +33,8 @@ class ShareViewController: SLComposeServiceViewController {
     } else {
       print("Error: \(NO_INFO_PLIST_URL_SCHEME_ERROR)")
     }
+    // Run didSelectPost automatically to avoid to launch pop up
+    Timer.scheduledTimer(timeInterval: 0.2, target: self, selector: #selector(self.didSelectPost), userInfo: nil, repeats: false)
   }
 
     override func isContentValid() -> Bool {
@@ -166,10 +168,6 @@ class ShareViewController: SLComposeServiceViewController {
         self.exit(withError: error.debugDescription)
         return
       }
-      guard let url = data as? URL else {
-        self.exit(withError: COULD_NOT_FIND_IMG_ERROR)
-        return
-      }
       guard let hostAppId = self.hostAppId else {
         self.exit(withError: NO_INFO_PLIST_INDENTIFIER_ERROR)
         return
@@ -181,18 +179,41 @@ class ShareViewController: SLComposeServiceViewController {
         return
       }
       
-      let mimeType = url.extractMimeType()
-      let fileExtension = url.pathExtension
-      let fileName = UUID().uuidString
-      let filePath = groupFileManagerContainer
-        .appendingPathComponent("\(fileName).\(fileExtension)")
+      if let uiImage = data as? UIImage {
+        let fileExtension = "png"
+        let fileName = UUID().uuidString
+        let filePath = groupFileManagerContainer
+          .appendingPathComponent("\(fileName).\(fileExtension)")
       
-      guard self.moveFileToDisk(from: url, to: filePath) else {
-        self.exit(withError: COULD_NOT_SAVE_FILE_ERROR)
+        guard let rawData = uiImage.pngData()
+        else {
+          self.exit(withError: "Error while getting raw data")
+          return
+        }
+        
+        guard FileManager.default.createFile(atPath: filePath.path, contents: rawData)
+        else {
+          self.exit(withError: "Error while createFile")
+          return
+        }
+        self.sharedItems.append([DATA_KEY: filePath.absoluteString, MIME_TYPE_KEY: "image/png"])        
+      } else if let url = data as? URL {
+        let mimeType = url.extractMimeType()
+        let fileExtension = url.pathExtension
+        let fileName = UUID().uuidString
+        let filePath = groupFileManagerContainer
+          .appendingPathComponent("\(fileName).\(fileExtension)")
+        
+        guard self.moveFileToDisk(from: url, to: filePath) else {
+          self.exit(withError: COULD_NOT_SAVE_FILE_ERROR)
+          return
+        }
+        
+        self.sharedItems.append([DATA_KEY: filePath.absoluteString, MIME_TYPE_KEY: mimeType])
+      } else {
+        self.exit(withError: COULD_NOT_FIND_IMG_ERROR)
         return
       }
-      
-      self.sharedItems.append([DATA_KEY: filePath.absoluteString, MIME_TYPE_KEY: mimeType])
       semaphore.signal()
     }
   }
