diff --git a/node_modules/react-native-create-thumbnail/README.md b/node_modules/react-native-create-thumbnail/README.md
index 4b949d5..8624e18 100644
--- a/node_modules/react-native-create-thumbnail/README.md
+++ b/node_modules/react-native-create-thumbnail/README.md
@@ -55,6 +55,7 @@ createThumbnail({
 | format    | `String` (default `jpeg`) | Thumbnail format, can be one of: `jpeg`, or `png`                         |
 | maxWidth  | `Number` (default `512`)  | Max thumbnail width in px                                                 |
 | maxHeight | `Number` (default `512`)  | Max thumbnail height in px                                                |
+| quality   | `Number` (default `90`)   | Thumbnail quality                                                         |
 | dirSize   | `Number` (default `100`)  | Maximum size of the cache directory (in megabytes). When this directory is full, the previously generated thumbnails will be deleted to clear about half of it's size.                        |
 | headers   |         `Object`          | Headers to load the video with. e.g. `{ Authorization: 'someAuthToken' }` |
 | cacheName   |         `String` (optional)          | Cache name for this thumbnail to avoid duplicate generation. If specified, and a thumbnail already exists with the same cache name, it will be returned instead of generating a new one. |
diff --git a/node_modules/react-native-create-thumbnail/android/build.gradle b/node_modules/react-native-create-thumbnail/android/build.gradle
index 7b97524..4834092 100644
--- a/node_modules/react-native-create-thumbnail/android/build.gradle
+++ b/node_modules/react-native-create-thumbnail/android/build.gradle
@@ -55,7 +55,7 @@ android {
         versionCode 1
         versionName "1.0"
     }
-    lintOptions {
+    lint {
         abortOnError false
     }
 }
@@ -77,7 +77,7 @@ repositories {
 
 dependencies {
     //noinspection GradleDynamicVersion
-    implementation 'com.facebook.react:react-native:+'  // From node_modules
+    implementation 'com.facebook.react:react-android:+'  // From node_modules
     implementation 'commons-io:commons-io:2.8.0'
 }
 
@@ -108,61 +108,3 @@ def configureReactNativePom(def pom) {
         }
     }
 }
-
-afterEvaluate { project ->
-    // some Gradle build hooks ref:
-    // https://www.oreilly.com/library/view/gradle-beyond-the/9781449373801/ch03.html
-    task androidJavadoc(type: Javadoc) {
-        source = android.sourceSets.main.java.srcDirs
-        classpath += files(android.bootClasspath)
-        project.getConfigurations().implementation.setCanBeResolved(true)
-        classpath += files(project.getConfigurations().getByName('implementation').asList())
-        include '**/*.java'
-    }
-
-    task androidJavadocJar(type: Jar, dependsOn: androidJavadoc) {
-        archiveClassifier = 'javadoc'
-        from androidJavadoc.destinationDir
-    }
-
-    task androidSourcesJar(type: Jar) {
-        archiveClassifier = 'sources'
-        from android.sourceSets.main.java.srcDirs
-        include '**/*.java'
-    }
-
-    android.libraryVariants.all { variant ->
-        def name = variant.name.capitalize()
-        task "jar${name}"(type: Jar, dependsOn: variant.javaCompileProvider) {
-            from variant.javaCompileProvider.get().destinationDir
-        }
-    }
-
-    artifacts {
-        archives androidSourcesJar
-        archives androidJavadocJar
-    }
-
-    publishing {
-        publications {
-            maven(MavenPublication) {
-                artifact androidSourcesJar
-            }
-        }
-    }
-}
-
-def myAttribute = Attribute.of("com.android.build.api.attributes.BuildTypeAttr", String)
-
-dependencies.attributesSchema {
-    // registers this attribute to the attributes schema
-    attribute(myAttribute)
-}
-
-configurations {
-    implementation {
-        attributes {
-            attribute(myAttribute, "release")
-        }
-    }
-}
diff --git a/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java b/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
index a0606c7..61f5cbf 100644
--- a/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
+++ b/node_modules/react-native-create-thumbnail/android/src/main/java/com/createthumbnail/CreateThumbnailModule.java
@@ -91,6 +91,7 @@ public class CreateThumbnailModule extends ReactContextBaseJavaModule {
         int timeStamp = options.hasKey("timeStamp") ? options.getInt("timeStamp") : 0;
         int maxWidth = options.hasKey("maxWidth") ? options.getInt("maxWidth") : 512;
         int maxHeight = options.hasKey("maxHeight") ? options.getInt("maxHeight") : 512;
+        int quality = options.hasKey("quality") ? options.getInt("quality") : 90;
         boolean onlySyncedFrames = options.hasKey("onlySyncedFrames") ? options.getBoolean("onlySyncedFrames") : true;
         HashMap headers = options.hasKey("headers") ? options.getMap("headers").toHashMap() : new HashMap<String, String>();
         String fileName = TextUtils.isEmpty(cacheName) ? ("thumb-" + UUID.randomUUID().toString()) : cacheName + "." + format;
@@ -101,11 +102,10 @@ public class CreateThumbnailModule extends ReactContextBaseJavaModule {
         file.createNewFile();
         fOut = new FileOutputStream(file);
 
-        // 100 means no compression, the lower you go, the stronger the compression
         if (format.equals("png")) {
-            image.compress(Bitmap.CompressFormat.PNG, 100, fOut);
+            image.compress(Bitmap.CompressFormat.PNG, quality, fOut);
         } else {
-            image.compress(Bitmap.CompressFormat.JPEG, 90, fOut);
+            image.compress(Bitmap.CompressFormat.JPEG, quality, fOut);
         }
 
         fOut.flush();
diff --git a/node_modules/react-native-create-thumbnail/index.d.ts b/node_modules/react-native-create-thumbnail/index.d.ts
index 3a61af8..58ed221 100644
--- a/node_modules/react-native-create-thumbnail/index.d.ts
+++ b/node_modules/react-native-create-thumbnail/index.d.ts
@@ -8,6 +8,7 @@ declare module "react-native-create-thumbnail" {
     cacheName?: string;
     maxWidth?: number;
     maxHeight?: number;
+    quality?: number;
     /**
      * iOS Only
      */
diff --git a/node_modules/react-native-create-thumbnail/ios/CreateThumbnail.m b/node_modules/react-native-create-thumbnail/ios/CreateThumbnail.m
index 4c87cfa..b8f18cf 100644
--- a/node_modules/react-native-create-thumbnail/ios/CreateThumbnail.m
+++ b/node_modules/react-native-create-thumbnail/ios/CreateThumbnail.m
@@ -14,6 +14,7 @@ RCT_EXPORT_METHOD(create:(NSDictionary *)config findEventsWithResolver:(RCTPromi
     NSDictionary *headers = config[@"headers"] ?: @{};
     CGFloat maxWidth = [config[@"maxWidth"] floatValue] ?: 512;
     CGFloat maxHeight = [config[@"maxHeight"] floatValue] ?: 512;
+    CGFloat quality = [config[@"quality"] floatValue] ?: 90;
     int timeToleranceMs = [config[@"timeToleranceMs"] intValue] ?: 2000;
 
     unsigned long long cacheDirSize = dirSize * 1024 * 1024;
@@ -61,7 +62,7 @@ RCT_EXPORT_METHOD(create:(NSDictionary *)config findEventsWithResolver:(RCTPromi
             if ([format isEqual: @"png"]) {
                 data = UIImagePNGRepresentation(thumbnail);
             } else {
-                data = UIImageJPEGRepresentation(thumbnail, 1.0);
+                data = UIImageJPEGRepresentation(thumbnail, quality / 100);
             }
 
             NSFileManager *fileManager = [NSFileManager defaultManager];
