diff --git a/node_modules/react-native-job-queue/android/build.gradle b/node_modules/react-native-job-queue/android/build.gradle
index fd4217e..4378548 100644
--- a/node_modules/react-native-job-queue/android/build.gradle
+++ b/node_modules/react-native-job-queue/android/build.gradle
@@ -65,6 +65,7 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     api 'com.facebook.react:react-native:+'
-    api 'android.arch.persistence.room:runtime:1.1.1'
-    annotationProcessor "android.arch.persistence.room:compiler:1.1.1"
+    implementation "androidx.room:room-runtime:2.5.2"
+    annotationProcessor "androidx.room:room-compiler:2.5.2"
+    implementation "androidx.room:room-ktx:2.5.2"
 }
